api/src/test/java/org/openmrs/module/fhir2/api/search/AllergyIntoleranceSearchQueryImplTest.java

```diff
@@ -0,0 +1,473 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.search;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.hasProperty;
+import static org.mockito.Mockito.when;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import ca.uhn.fhir.rest.api.server.IBundleProvider;
+import ca.uhn.fhir.rest.param.ReferenceAndListParam;
+import ca.uhn.fhir.rest.param.ReferenceOrListParam;
+import ca.uhn.fhir.rest.param.ReferenceParam;
+import ca.uhn.fhir.rest.param.TokenAndListParam;
+import ca.uhn.fhir.rest.param.TokenOrListParam;
+import ca.uhn.fhir.rest.param.TokenParam;
+import org.hibernate.SessionFactory;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.AllergyIntolerance;
+import org.hl7.fhir.r4.model.Patient;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.openmrs.Allergy;
+import org.openmrs.module.fhir2.FhirConstants;
+import org.openmrs.module.fhir2.TestFhirSpringConfiguration;
+import org.openmrs.module.fhir2.api.FhirGlobalPropertyService;
+import org.openmrs.module.fhir2.api.dao.FhirAllergyIntoleranceDao;
+import org.openmrs.module.fhir2.api.dao.impl.FhirAllergyIntoleranceDaoImpl;
+import org.openmrs.module.fhir2.api.search.param.SearchParameterMap;
+import org.openmrs.module.fhir2.api.translators.AllergyIntoleranceTranslator;
+import org.openmrs.module.fhir2.api.translators.impl.AllergyIntoleranceSeverityTranslatorImpl;
+import org.openmrs.test.BaseModuleContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+
+@ContextConfiguration(classes = TestFhirSpringConfiguration.class, inheritLocations = false)
+public class AllergyIntoleranceSearchQueryImplTest extends BaseModuleContextSensitiveTest {
```

Comment:
Can you add 2 or more tests, which test for complex queries i.e 
1. combines two or more search parameters e.g by patientUuid(subject), severity, clinical status
2. two or more plus sort

Reply:
Sure @corneliouzbett . Can we also change the access levels of getters and setters in all Dao classes because we might not get access in the search query package ?

Reply:
@ibacher  Can you take a look at my comment as well ?

Reply:
@varung-31 What do you need access to that you don't already have access to from the search query? We want to keep the public interfaces as minimal as we can.

Reply:
@ibacher  Sorry, I forgot to mention. I am having a problem mocking some objects in the test classes because of restricted access, like what happened with mocking GlobalPropertyService in case of AllergyIntolerance resource. I had to change the translator and BaseFhirDao setter access to PUBLIC.

Reply:
@varung-31 I see. I think there's a cleaner way to handle this, which I've committed to this PR. The `FhirGlobalPropertyService` is (hopefully)  bit of a special case.

Reply:
Thanks @ibacher

Reply:
The build is failing because the module is unable to connect to openmrs database. Any idea why that is happening?
The tests are working locally, but on `mvn clean package`, this build failure pops up.